docs/src/man/doctests.md

```diff
@@ -105,59 +102,102 @@ line where checking should stop, i.e.
 julia> div(1, 0)
 ERROR: DivideError: integer division error
 [...]
+```
+````
+
+## Preserving definitions between blocks
+
+Every doctest block is evaluated inside its own `module`. This means that definitions
+(types, variables, functions etc.) from a block can *not* be used in the next block.
+A doctest block can be labeled in order to collect several blocks into the same module.
+All blocks with the same label (in the same file) will be evaluated in the same module,
+and hence share scope. This can be useful if the same definitions are used in more than
+one block, with for example text, or other doctest blocks, in between. Example:
+
+````markdown
+```jldoctest mylabel
+julia> type Foo
+            x
+            y
+       end
+```
+
+Some text discussing the `Foo` type...
+In the next doctest block the type `Foo` can be used without redefining:
 
+```jldoctest mylabel
+julia> foo = Foo(1, 2)
+Foo(1,2)
+```
+
+Some more text...
+In the next block the variable `foo` can be used:
+
+```jldoctest mylabel
+julia> foo.x
+1
+
+julia> foo.y
+2
 ```
 ````
 
-## Skipping Doctests
+!!! note
 
-Doctesting can be disabled by setting the [`makedocs`](@ref) keyword `doctest = false`.
-This should only be done when initially laying out the structure of a package's
-documentation, after which it's encouraged to always run doctests when building docs.
+    Labeled doctest blocks does not need to be consecutive (as in the example above) to be
+    included in the same module. They can be interspaced with unlabeled blocks or blocks
+    with another label.
 
 ## Setup Code
 
 Doctests may require some setup code that must be evaluated prior to that of the actual
-example, but that should not be displayed in the final documentation. It could also be that
-several separate doctests require the same definitions. For both of these cases a `@meta`
-block containing a `DocTestSetup = ...` value can be used as follows:
+example, but that should not be displayed in the final documentation. For this purpose a
+`@meta` block containing a `DocTestSetup = ...` value can be used as follows:
 
-    ```jldoctest
-    julia> using DataFrames
-
-    julia> df = DataFrame(A = 1:10, B = 2:2:20);
+````markdown
+Some text discussing a function `foo`...
 
-    ```
+The definition of `foo` is defined in a `@meta` block below, and thus not visible in the
+rendered documentation:
 
-    Some text discussing `df`...
+```@meta
```

Comment:
Could also be a bit more concise maybe. I might keep only one definition and one doctest block.